migrator/migrations/m_109_to_m_110_replace_deprecated_resources/migration.go

```diff
@@ -0,0 +1,134 @@
+package m109tom110
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/tecbot/gorocksdb"
+)
+
+const (
+	batchSize = 500
+)
+
+// Replacement resources
+const (
+	Access              = "Access"
+	DeploymentExtension = "DeploymentExtension"
+	Image               = "Image"
+	Integration         = "Integration"
+)
+
+// Replaced resources
+const (
+	APIToken             = "APIToken"
+	AuthProvider         = "AuthProvider"
+	BackupPlugins        = "BackupPlugins"
+	Group                = "Group"
+	ImageComponent       = "ImageComponent"
+	ImageIntegration     = "ImageIntegration"
+	Indicator            = "Indicator"
+	Licenses             = "Licenses"
+	NetworkBaseline      = "NetworkBaseline"
+	Notifier             = "Notifier"
+	ProcessWhitelist     = "ProcessWhitelist"
+	Risk                 = "Risk"
+	SignatureIntegration = "SignatureIntegration"
+	User                 = "User"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 109,
+		VersionAfter:   storage.Version{SeqNum: 110},
+		Run: func(databases *types.Databases) error {
+			return migatePermissionSets(databases.RocksDB)
+		},
+	}
+
+	prefix = []byte("permission_sets")
+
+	replacements = map[string]string{
+		APIToken:             Integration,
+		AuthProvider:         Access,
+		BackupPlugins:        Integration,
+		Group:                Access,
+		ImageComponent:       Image,
+		ImageIntegration:     Integration,
+		Indicator:            DeploymentExtension,
+		Licenses:             Access,
+		NetworkBaseline:      DeploymentExtension,
+		Notifier:             Integration,
+		ProcessWhitelist:     DeploymentExtension,
+		Risk:                 DeploymentExtension,
+		SignatureIntegration: Integration,
+		User:                 Access,
+	}
+
+	readOpts  = gorocksdb.NewDefaultReadOptions()
+	writeOpts = gorocksdb.NewDefaultWriteOptions()
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+func propagatePermission(resource string, accessLevel storage.Access, permissions map[string]storage.Access) storage.Access {
+	if _, found := permissions[resource]; !found {
+		return accessLevel
+	}
+	oldLevel := permissions[resource]
+	if accessLevel > oldLevel {
+		return oldLevel
+	}
+	return accessLevel
```

Comment:
General question on this: have we decided that this is the way to go? If we keep as-is, this _could_ potentially break existing customer's permission sets after migration. 

Say customer has a permission set with write to `AuthProvider` and read to `User`, we will have after migration read to `Access`, which would break his access to `AuthProvider` (which he seemingly required, since he set it to begin with). 

Is this something that we have decided is OK for customers / will there be a special announcement and we expect them to manually change this afterwards?

I might miss context, but haven't found anything in the related slack channel.

Reply:
Hm, I see.
I would have gone for max due to it _probably_ not resulting in directly visible breaking changes (i.e. receiving a support case due to user X not being able to create image integrations), but min is also fine since its a breaking change either way.
My point of emphasis would be to _specifically highlight this change_, in order for us to not get too many support cases beginning with the next release.

Reply:
> My point of emphasis would be to specifically highlight this change, in order for us to not get too many support cases beginning with the next release.

This is fair, Daniel.